pettingzoo/utils/env.py

```diff
@@ -22,7 +18,7 @@
 """
 
 
-class AECEnv:
+class AECEnv(Generic[AgentID, ObsType, ActionType]):
```

Comment:
My guess is if you have to pass in ObsType or ActionType when you define an AECEnv it will make things a bit more confusing, kind of feel like it would be simplest to keep it as it was before but just replace uses of `str` in signatures to `AgentID` (I've already tried to do this a few places because it's cleaner)

Reply:
What inheriting from Generic does is it makes the class have the ability to accept type parameters when you define a child class. When instantiating the class, the type parameters are inferred.

However, providing type parameters is purely optional. You don't have to pass in an AgentID, ObsType or ActionType when defining an AECEnv unless you want to. If you don't provide these bits of info, then they are left as unknown. This means that the change is fully backward-compatible. 

I.E, if the user does not touch their code after this change, it will still work, and it will not introduce any new type errors.

Reply:
Also, Gymnasium also uses Generics when defining their environment: https://github.com/Farama-Foundation/Gymnasium/blob/main/gymnasium/core.py#L21

Reply:
Oh I see, well in that case this looks good then, thanks for the heads up about Gymnasium doing it as well